modules/lib/lib-portal/src/main/resources/lib/xp/portal.ts

```diff
@@ -471,9 +471,10 @@ interface GetCurrentComponentHandler<Config extends object = object,
  *
  * @returns {object|null} The current component as JSON.
  */
-export function getComponent<Config extends object = object,
+export function getComponent<
+    Config extends NestedRecord = NestedRecord,
     Regions extends Record<string, Region> = Record<string, Region>,
-    >(): Component<Config, Regions> | null {
+>(): Component<Config, Regions> | null {
```

Comment:
I suggest we let this function take 1 type parameter that extends  `Component`. This would be consistent with how we work with content.

```typescript
export function getComponent<
    _Component extends Component = Component
>(): _Component | null {
```

Reply:
I agree, it's a breaking change though.

Reply:
I suggest you postpone this until XP 8 types then.